ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardChain.java

```diff
@@ -35,20 +36,48 @@
 public class BackwardChain {
   private static final Logger LOG = getLogger(BackwardChain.class);
 
+  private static final String FIRST_STORED_ANCESTOR_KEY = "firstStoredAncestor";
+  private static final String LAST_STORED_PIVOT_KEY = "lastStoredPivot";
+
   private final GenericKeyValueStorageFacade<Hash, BlockHeader> headers;
   private final GenericKeyValueStorageFacade<Hash, Block> blocks;
   private final GenericKeyValueStorageFacade<Hash, Hash> chainStorage;
-  private Optional<BlockHeader> firstStoredAncestor = Optional.empty();
-  private Optional<BlockHeader> lastStoredPivot = Optional.empty();
+  private final GenericKeyValueStorageFacade<String, BlockHeader> variablesStorage;
+  private Optional<BlockHeader> firstStoredAncestor;
+  private Optional<BlockHeader> lastStoredPivot;
   private final Queue<Hash> hashesToAppend = new ArrayDeque<>();
 
   public BackwardChain(
       final GenericKeyValueStorageFacade<Hash, BlockHeader> headersStorage,
       final GenericKeyValueStorageFacade<Hash, Block> blocksStorage,
-      final GenericKeyValueStorageFacade<Hash, Hash> chainStorage) {
+      final GenericKeyValueStorageFacade<Hash, Hash> chainStorage,
+      final GenericKeyValueStorageFacade<String, BlockHeader> variablesStorage) {
```

Comment:
Bws uses the db to keep some data during the sync, for the primary purpose that it could fairly large and not fit in memory. 
But I think that supporting resuming after the restart was not the intent of the storage, because some info were missing to fully support resuming, specifically 2 variables that point to the current status of the bws session.
So this `variablesStorage` is used to store the status of the current bws session, so resuming is fully supported.

Reply:
Got it. Maybe sessionDataStorage is a clearer name? Variables seems quite generic to me.